src/shared_utilities/models.jl

```diff
@@ -145,6 +146,46 @@ function make_update_boundary_fluxes(model::AbstractModel)
     return update_boundary_fluxes!
 end
 
+"""
+    make_update_drivers(model::AbstractModel)
+
+Return an `update_drivers!` function that updates the cache parameters in `p`
+corresponding to forcing at the surface.
+"""
+function make_update_drivers(model::AbstractModel)
+    function update_drivers!(p, Y, t) end
+    return update_drivers!
+end
+
+"""
+     make_update_cache(model::AbstractModel)
+
+A helper function which updates all cache variables of a model;
+currently only used in `set_initial_cache` since not all 
+cache variables are updated at the same time.
+"""
+function make_update_cache(model::AbstractModel)
+    update_drivers! = make_update_drivers(model)
+    update_aux! = make_update_aux(model)
+    update_boundary_fluxes! = make_update_boundary_fluxes(model)
+    function update_cache!(p, Y, t)
+        update_drivers!(p, Y, t)
+        update_aux!(p, Y, t)
+        update_boundary_fluxes!(p, Y, t)
```

Comment:
you'e right! but I think this will be ok, because for any integrated model, we have a `make_update_aux` method which calls `update_aux` for each of the components, and for likewise for `boundary_fluxes`. so instead of 
```
make_update_cache(land_model)
        uc1! = make_update_cache(land_model.component1)
        uc2! = make_update_cache(land_model.component2)
       function uc!(p,Y,t)
                uc1!(p,Y,t)
                uc2!(p,Y,t)
      end
return uc!
end
```
which would have the issue you describe, we have a method of `make_update_aux(land_model)` which does something like the above, and similar for `boundary_fluxes` and then those can be combined in `update_cache` using the code you commented on.

Reply:
Good point! That sounds like a really nice setup